ksql-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4

```diff
@@ -294,6 +294,7 @@ identifier
     | nonReserved            #unquotedIdentifier
     | BACKQUOTED_IDENTIFIER  #backQuotedIdentifier
     | DIGIT_IDENTIFIER       #digitIdentifier
+    | KEY                    #keyLiteral
```

Comment:
`KEY` is not an identifier!

Reply:
`KEY` _is_ a valid identifier.  It's valid to have a property, column, stream or table called `KEY`.  We have a with clause property called `KEY`.  Not allowing `KEY` as an identifier would be a breaking change, right?

Reply:
Sure, the question is why should we have it defined here? Isn't is already covered by `IDENTIFIER`?

Reply:
Sure, the question is why should we have it defined here? Isn't is already covered by `IDENTIFIER`?

Reply:
@big-andy-coates did you see the above comment?

Reply:
The lexer converts the text string `KEY` to `KEY` not to an `IDENTIFIER. So if we want the parser to accept `KEY` as an `identifier` then we need it here.

Reply:
The problem is now with this change `KEY` is a keyword similar to `SELECT` for instance. So if users needs to use it as an identifier name, they should use quoted identifier which is not supported yet. SO we should not have it as an identifier.
This would be a breaking change since we won't be able to use `KEY` in the `WITH` clause as we have been using so far.

Reply:
There are tests in this PR that prove this is not the case. You can have `KEY` as a field name: https://github.com/confluentinc/ksql/pull/3034/files#diff-bd64251df6ac4d5143a0104519761eadR100

Reply:
Regarding using `KEY` in the `WITH` clause... given the all the tests are passing, I think you can assume it still can be used in the `WITH` clause.

Reply:
OK, so to close this out... chatting to @hjafarpour  offline and his issue was that conceptually `KEY` should be a keyword and therefore not allowed as an unquoted identifier.  I've switched it to be `nonReserved` and @hjafarpour is now, hopefully, happy.